app/src/org/commcare/dalvik/activities/CommCareHomeActivity.java

```diff
@@ -270,152 +389,169 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
+                            //an empty state, Go home!
+                            currentState.reset();
+                            uiController.refreshView();
+                            return;
+                        } else {
+                            currentState.getSession().stepBack();
+                        }
+                    } else if (resultCode == RESULT_OK) {
+                        CommCareSession session = currentState.getSession();
+                        if (sessionStateUnchangedSinceCallout(session, intent)) {
+                            //Get our command, set it, and continue forward
+                            String command = intent.getStringExtra(SessionFrame.STATE_COMMAND_ID);
+                            session.setCommand(command);
+                        } else {
+                            clearSessionAndExit(currentState, true);
+                            return;
+                        }
+                    }
+                    break;
+                case GET_CASE:
+                    //TODO: We might need to load this from serialized state?
+                    AndroidSessionWrapper asw = CommCareApplication._().getCurrentSessionWrapper();
+                    CommCareSession currentSession = asw.getSession();
+                    if (resultCode == RESULT_CANCELED) {
+                        currentSession.stepBack();
+                    } else if (resultCode == RESULT_OK) {
+                        if (sessionStateUnchangedSinceCallout(currentSession, intent)) {
+                            String sessionDatumId = currentSession.getNeededDatum().getDataId();
+                            String chosenCaseId = intent.getStringExtra(SessionFrame.STATE_DATUM_VAL);
+                            currentSession.setDatum(sessionDatumId, chosenCaseId);
+                        } else {
+                            clearSessionAndExit(asw, true);
+                            return;
+                        }
+                    }
+                    break;
+                case MODEL_RESULT:
+                    boolean fetchNext = processReturnFromFormEntry(resultCode, intent);
+                    if (!fetchNext) {
+                        return;
+                    }
+                    break;
+                case AUTHENTICATION_FOR_PIN:
+                    if (resultCode == RESULT_OK) {
+                        String password = intent.getStringExtra(PinAuthenticationActivity.PASSWORD_FROM_AUTH);
+                        launchPinCreateScreen(LoginMode.PASSWORD, password);
+                    }
+                    return;
+                case CREATE_PIN:
+                    boolean choseRememberPassword = (intent == null) ? false :
+                            intent.getBooleanExtra(CreatePinActivity.CHOSE_REMEMBER_PASSWORD, false);
+                    if (choseRememberPassword) {
+                        CommCareApplication._().closeUserSession();
```

Comment:
We're triggering a log out, not log in. The user selected the "remember password for next login" option, which saves the password and then logs you out.